config/v1/types_oauth.go

```diff
@@ -542,6 +547,29 @@ type OpenIDClaims struct {
 	Email []string `json:"email"`
 }
 
+// masterAuthConfig configures authentication options for the kube api server
+type MasterAuthConfig struct {
+	// oauthMetadata contains the discovery endpoint data for OAuth 2.0
+	// Authorization Server Metadata for an external OAuth server.
+	// See IETF Draft:
+	// https://tools.ietf.org/html/draft-ietf-oauth-discovery-04#section-2
+	OAuthMetadata ConfigMapReference `json:"oauthMetadata"`
+
+	// webhookTokenAuthenticators configures remote token reviewers
+	WebhookTokenAuthenticators []WebhookTokenAuthenticator `json:"webhookTokenAuthenticators"`
```

Comment:
I think this piece only belongs in spec.

Reply:
It would set the operator based on the route for osin right?